internal/helm-storage-backend/release.go

```diff
@@ -2,32 +2,206 @@ package helmstoragebackend
+	default:
+		return nil, h.internalError(errors.Wrap(err, "while getting Helm release"))
+	}
+
+	releaseData := ReleaseDetails{
+		Name:      release.Name,
+		Namespace: release.Namespace,
+		Chart: ChartDetails{
+			Name:    release.Chart.Metadata.Name,
+			Version: release.Chart.Metadata.Version,
+			Repo:    releaseContext.ChartLocation,
+		},
+	}
+
+	value, err := json.Marshal(releaseData)
+	if err != nil {
+		return nil, errors.Wrap(err, "while marshaling response value")
+	}
+
 	return &pb.GetValueResponse{
-		Value: []byte(`{"handler": "release"}`),
+		Value: value,
 	}, nil
 }
+
+// OnUpdate checks whether a given Helm release is accessible this storage backend.
+func (h *ReleaseHandler) OnUpdate(_ context.Context, req *pb.OnUpdateRequest) (*pb.OnUpdateResponse, error) {
+	if err := h.checkIfHelmReleaseExist(req.TypeInstanceId, req.Context); err != nil {
+		return nil, err
+	}
+	return &pb.OnUpdateResponse{}, nil
+}
+
+// OnDelete is NOP. Currently, we are not sure whether the release should be deleted or this should be more an information
+// that someone wants to deregister this TypeInstance.
+func (h *ReleaseHandler) OnDelete(_ context.Context, _ *pb.OnDeleteRequest) (*pb.OnDeleteResponse, error) {
+	return &pb.OnDeleteResponse{}, nil
+}
+
+// GetLockedBy is NOP.
+func (h *ReleaseHandler) GetLockedBy(_ context.Context, _ *pb.GetLockedByRequest) (*pb.GetLockedByResponse, error) {
+	return &pb.GetLockedByResponse{}, nil
+}
+
+// OnLock is NOP.
+func (h *ReleaseHandler) OnLock(_ context.Context, _ *pb.OnLockRequest) (*pb.OnLockResponse, error) {
+	return &pb.OnLockResponse{}, nil
+}
+
+// OnUnlock is NOP.
+func (h *ReleaseHandler) OnUnlock(_ context.Context, _ *pb.OnUnlockRequest) (*pb.OnUnlockResponse, error) {
+	return &pb.OnUnlockResponse{}, nil
+}
+
+func (h *ReleaseHandler) getReleaseContext(contextBytes []byte) (*ReleaseContext, error) {
+	var ctx ReleaseContext
+	err := json.Unmarshal(contextBytes, &ctx)
+	if err != nil {
+		return nil, h.internalError(errors.Wrap(err, "while unmarshaling context"))
+	}
+
+	if ctx.Driver == nil {
+		ctx.Driver = ptr.String(defaultHelmDriver)
+	}
+
+	return &ctx, nil
+}
+
+func (h *ReleaseHandler) checkIfHelmReleaseExist(ti string, ctx []byte) error {
+	releaseContext, err := h.getReleaseContext(ctx)
+	if err != nil {
+		return err
+	}
+
+	helmGet, err := h.newHelmGet(h.helmCfgFlags, *releaseContext.Driver, releaseContext.Namespace)
+	if err != nil {
+		return h.internalError(errors.Wrap(err, "while creating Helm get release client"))
+	}
+
+	_, err = helmGet.Run(releaseContext.Name)
+	switch {
+	case err == nil:
+	case errors.Is(err, driver.ErrReleaseNotFound):
+		return status.Error(codes.NotFound, fmt.Sprintf("Helm release '%s/%s' for TypeInstance '%s' was not found", releaseContext.Namespace, releaseContext.Name, ti))
+	default:
+		return h.internalError(errors.Wrap(err, "while checking if Helm release exists"))
+	}
```

Comment:
This is almost identical to the snippet from `GetValue`:

```go
	release, err := helmGet.Run(releaseContext.Name)
	switch {
	case err == nil:
	case errors.Is(err, driver.ErrReleaseNotFound):
		return nil, status.Error(codes.NotFound, fmt.Sprintf("Helm release '%s/%s' for TypeInstance '%s' was not found", releaseContext.Namespace, releaseContext.Name, req.TypeInstanceId))
	default:
		return nil, h.internalError(errors.Wrap(err, "while getting Helm release"))
	}
```

Consider having it defined only once, as it would be easier to maintain.

Reply:
I thought about it, but it also complicates this function. However, I "merged" it as suggested.